packages/core/src/connectivity/scp-cf/jwt.ts

```diff
@@ -146,23 +146,24 @@ export async function verifyJwt(
   return fetchAndCacheKeyAndVerify(creds, verificationKeyURL, token, options); // Verify only here
 }
 
-function fetchAndCacheKeyAndVerify(
+async function fetchAndCacheKeyAndVerify(
   creds: XsuaaServiceCredentials,
   verificationKeyURL: string,
   token: string,
   options?: VerifyJwtOptions
-) {
-  return getVerificationKey(creds, verificationKeyURL)
-    .catch(error => {
-      throw new ErrorWithCause(
-        'Failed to verify JWT. Could not retrieve verification key.',
-        error
-      );
-    })
-    .then(key =>
-      options ? cacheVerificationKey(verificationKeyURL, key, options) : key
-    )
-    .then(key => verifyJwtWithKey(token, key.value));
+): Promise<JwtPayload> {
+  try {
+    const key = await getVerificationKey(creds, verificationKeyURL);
+    if (options?.cacheVerificationKeys) {
+      verificationKeyCache.set(verificationKeyURL, key);
+    }
+    return verifyJwtWithKey(token, key.value);
+  } catch (error) {
+    throw new ErrorWithCause(
+      'Failed to verify JWT. Could not retrieve verification key.',
+      error
+    );
+  }
```

Comment:
This is different behavior than before. The error message does not fit the error cause.

```suggestion
    const key = await getVerificationKey(creds, verificationKeyURL).catch(error => {
      throw new ErrorWithCause(
        'Failed to verify JWT. Could not retrieve verification key.',
        error
      );
    });
    if (options?.cacheVerificationKeys) {
      verificationKeyCache.set(verificationKeyURL, key);
    }
    return verifyJwtWithKey(token, key.value);
  }
```